tasks/docs.rake

```diff
@@ -0,0 +1,8 @@
+desc 'Generate the API documentation.'
+task(:docs) do
+  env = {}
+  ENV['DOCSTRINGS'] = '1'
+  ENV['SITEMAP_BASEURL'] = 'http://docs.aws.amazon.com/sdk-for-ruby/v3/api/'
```

Comment:
I think we want to avoid putting the internals of the docs into the SDK repo.  
We do provide docs tasks for some of our other (non-sdk) libraries, but expect those ENV vars to be set.  Eg: [aws-sdk-ruby-release-tools: docs.rake](https://github.com/aws/aws-sdk-ruby-release-tools/blob/6475953e6c531cc2c494a7dc9987a1feff199414/docs.rake)